collect_app/src/main/java/org/odk/collect/android/utilities/SQLiteQueryBuilder.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.odk.collect.android.utilities;
+
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteException;
+
+public class SQLiteQueryBuilder {
```

Comment:
I think this name is confusing because it's the same as https://developer.android.com/reference/android/database/sqlite/SQLiteQueryBuilder.html

Honestly, I would tend to just do the queries inline so that it's super explicit what is going on. If you really want to introduce this abstraction, I think it should come with tests! It's relatively easy to test and it's also relatively easy to introduce really annoying bugs in something like this.

Reply:
I find it useful and clear so I added tests since I would like to keep it.